### PR TITLE
Update deprecated notice 

### PR DESCRIPTION
**Pull request by cherifGsoul**
This adds deprecated notice on and links to the features that replaces those deprecations.

#336

### CODE REVIEW

File: docs/beforeremove.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 @typedef {function} can-component/beforeremove beforeremove
 @parent can-component.deprecated
 
+@deprecated Use [can-component/connectedCallback] instead.
```
Comment (chasenlehara) [inline docs/beforeremove.md]: It’d be good to link to the part of the migration guide where this is covered: “inserted/removed event” in https://canjs.com/doc/migrate-4.html

That section doesn’t have an ID you can easily link to, but I think it’d be good if it did. Right now you can increase the outline for that page to make it 1) display more in the TOC and 2) generate the required IDs. I think something like `@outline 2` on that page would do the trick, but you’ll need to check (and open a PR).

Also, I think you can say this was deprecated in 4.0

File: docs/events.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
 @property {Object} can-component.prototype.events events
 @parent can-component.deprecated
 
+@deprecated Use [can-control] instance on the component custom tag instead.
```
Comment (chasenlehara) [inline docs/events.md]: I’m not 100% sure what this means… can more info be provided? Providing more info inline would be great, or even a link to a CodePen would be useful.

File: docs/helpers.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 @property {Object.<String,can-stache.helper>} can-component.prototype.helpers helpers
 @parent can-component.deprecated
 
+@deprecated Use [can-component.prototype.ViewModel] methods instead.
```
Comment (chasenlehara) [inline docs/helpers.md]: I understand what’s being said here, but a quick example would go a long way to helping others understand.

File: docs/leakscope.md [hunk 4]
```diff
@@ -1,6 +1,8 @@
 @property {Boolean} can-component.prototype.leakScope leakScope
 @parent can-component.deprecated
 
+@deprecated Pass your values instead.
```
Comment (chasenlehara) [inline docs/leakscope.md]: Same; examples are helpful to people upgrading / cleaning up tech debt. 😊

File: docs/view-model.md [hunk 5]
```diff
@@ -1,6 +1,8 @@
 @function can-component.prototype.view-model viewModel
 @parent can-component.deprecated
 
+@deprecated Use [can-component.prototype.ViewModel] methods instead.
```
Comment (chasenlehara) [inline docs/view-model.md]: Same thing here.

File: docs/leakscope.md (hunk 4 above)
Comment (cherifGsoul) [reply docs/leakscope.md]: I already fixed that

File: docs/beforeremove.md (hunk 1 above)
Comment (cherifGsoul) [reply docs/beforeremove.md]: Done!

File: docs/events.md (hunk 2 above)
Comment (cherifGsoul) [reply docs/events.md]: Done!

File: docs/leakscope.md (hunk 4 above)
Comment (cherifGsoul) [reply docs/leakscope.md]: Done!

File: docs/view-model.md (hunk 5 above)
Comment (cherifGsoul) [reply docs/view-model.md]: Done!

**Pull request merged**